test/simulation/lntest/node.go

```diff
@@ -76,6 +82,15 @@ var (
 func generateListeningPorts() (int, int, int) {
 	var p2p, rpc, rest int
 	if numActiveNodes == 0 {
+	    freePorts, err := getFreePorts(3)
+		if err != nil {
+		    fmt.Println(fmt.Errorf("could not find free ports %v", err))
```

Comment:
You should return the error here, and not merely print it (method signature need to be changed to `int, int, int, error` which is a bit weird, but probably fine).